### ISSUE TITLE
Changing editor does not create new object with correct value

### ISSUE THREAD

**Issue by totszwai** [json-editor/json-editor#1441]
### General information

- json-editor version: 

### Expected behavior

When changing a selected value that is pointing to a $ref, the json output should either
- rebuild the json object completely OR
- at least partially rebuild the sub object 

In the example, switching between the 2 refs does not correctly update the final JSON. Existing fields were not cleared only new field(s) were added/removed.

### Actual behavior

The JSON object being output contains incorrect value.

### Steps to reproduce the behavior

1. Go to the [example](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsihelgDQgTliHQ5VYBmYcrgAmWADZY6qJqxCZRA/OnFQcAN3zjmDWW0L4AHjjHQAFqgAMbGmQQ5EUUrkJgxqAIxsAxvmwOsJBhYdVwOSCxyKABPAPxybwtGcmY2GG8EcWpUEEF4JwB3ei4sAFYQNjAqACssbzVxfGiwWhz4IQNxCpBERKwDVFAAEl6zfvwcszIIRBQAejnqxAQAWlHxgDohAHM5kXJ8fig5gCZLM5WPE7n1gbYYiCwcqtr67tgoSRyABQBpAEkAAQAYS4EHw8Dg8EBAEFvv9uuFHlEYPRBiBhABHGAidEPJ4oEAwfJYbaRbpifjKVSeAC+92ijzxjIJPQoxO2FKUKigOSxOl59JAxEQiHwZOZTMJLzqvPuwS+hIAsvQxRK2BDogB5fioADaw2EusJAGI9gJicEoYg5vz6FAbXxtsSVhQIWL6lCQPTDQIcmbKZbYAgbXbHDaxN5yIyoCtyGLvSxfcaQAGLZDg0hbVhMQKI3Vo9AVnVvQBdWlCwMZ63o/CekOS1mOSjwTlsQLoQj6kBOl1upB1zNcqMxkDlnxcVzwdFiXqUaBewmgwhTwEAKQAylqAHKAmX1QHN9D1dDCZD3A4D+vTtAgfHPGqy7r8mDCXEoPUgXzkEQ4sdsJFIlgNFb2/X931Ae9CWJOgyXILk5xgBcEByABNFpAQyFQRD3KgnGJQEoDGTD4hwnFASoaISPELJW0BLJHBWMDkArIUAGtEAgRscmbDlExANisGiTiS1vKDwEfN4AM4ZFgIYSCWW4wleNbboOy7D8MQTdtvDHIVEBgAAvAkFKlIkSTg/jvFPUIlLZFs2xAWcoyQodCSmKAZnmOZJHieANkIdJOGWI4NgyQg5kCFZMDmApiREMAChtOL4AAZmubBvG8Vs5my7YVgIYgRGLGjXPSJibLRdt4E7btMu+E5SgANjYjxukyiBGqa8g2rYDrUoADgAFh69q6m+UoTg8UbywrNhI0LTM4zFFZwQOQh5LvRTpUkuV2BkoDUU27xDograzJUxzKWpXlCQAEQ3fhqnIABVbYAAkNzMABRCAIGBGA13+RAhpe/gAGoACUCjXX4ACEwDMAwABUAE4ADF8HR6p4BOaoAEURCGgBeBCXOQm8QGBQ7+HSZxAX4ThCEwmm6boHB43wDYICuKytE5MTtvsvj5u5GlCRhLqAHFgSVfi2P5uzLq5KkeRySGNxhFYtRhb7vhWLqrKQsYoiwQxbtMpt2VU0XVfFjEABk2MhoTtlJoUnAFy2eOtq6xdukBtlQgAtRB8fgR78FQh2QagQywClk5thEKWYWJ922HaeATqV32VZunIAHUriwMAXqgVGuHxmAN1Qpqhv4wIR2gLARBwE62Nzhz87VwkADVUsQLVC5gdB1AKaJvu2JqwDhkQN1KeB1EMsx3rXSHIaGwh3vx9H8Zejc+8LqgXb7gpUZhMAThOchLEIQvMThfAlWaDd/iatdwYdywXrhoaCm8CcF6kN3rqBEGxVCVAYRUilvgb43h/jozmEcABhlLAAA0oBsUIOgVKg1SgDSJqlfGcwzDwEICIOGlgPAwmiOjJqhkqDbiwEqIahh0aGQ8CvdAqJMbMJgE1LUqUlSECloQVKQ1tw9W+r8AoQ0+53RgA7Iad13qF0JqhEQqN8b4Dht4Qga4pbvQKFLQyfdvCoRek1d6bF0DvWRvAYE/B0FgEsJYVCyNkaoUsJicQyMqCWGBJiQy6MRCNCwFQKgDtvrfWiPAB2G4WiWGqCcb6A1LDowAOzoxhvjKWfdtziGPonLUhhxDBxEKlQwbF0b8EIFcE4qMNyEEougTJHhqhcGBN8BRk1MQeHgGuCAZh0F9w8MHYEkM2LVAdh4KA4MIB3REFAfG3hg6ZPQSIUohdqjdQ3ANb4BkTham8JiAa1QqCEAGkqKgIgmovV+KUQyY9qiGWRg7LG3g3buyFC+N83YToolxD4RWbAFbiEct4Y2QEzZ7U9t0LOOdqrNzoG3DuelRYoqhMWbwq14hEE2uJfce1AIohAqAQFsAzriWVrbAuhJMSpSahAdGcMebI0MO9LUA04aQ1KHMVKmI2LQEsOgCAA1UKYianDDchlDJNLABuVKrVyAQG+qTUWiEKY5GpiiWmvg6AMyZoRYidBHBdW5rzIUvgIVdxFk5f2ORJbNRlnLdiitBYXTznS3uIBvrAlUSsb6G4jbDJhebO1NsHV2wDqjKZJwmrAnQMCYOwc7qoWIb8Fi9xxSRr9jGnIa4HYDX4FgAamITjRJtKlMAvx8BzDjZDQgBR078URSZc6Vtu4+vtuDMwiBLDwH4ANb6EKtSZIHt9cGjcICd09ayYliIDpks2mxGIeae7239VZcg6gN09oDvrQ2QpDD7ujfSqm30+7bExJPdBmTDAvTAPjVGHhvBNVSsCIa4gXpDQdoZRAmI+6ICVIXO6hBIY2iGtK+AcwNxKgziAaIZ7rq+uRogGE70czHD7nDVKqNvpgH4N9cg31MS2j7k1TEWBLDfRUZk7eBQCiEHQUNLU4glS/Flj8uaGIcw8P+ZpSlf4QW2rBaCr80LTYRpzY5dtalZ16V+fx+0m1nLzjciAF2uZ7SAhFGqLA64ty7mJYeJgJ4zzdF7PAV0l4PRuUggqVkcNSQEWRnZwcXoGRmUXdJCIK7azXnREMI0/pzS02rCGOYnmGwHpyNZ2z7oYvTiFGtIgUhIibRC36U04WgzWlypOBAcx+z2a9Lxv5rduzJcRPi4gdB4wYodSi5aEwxJOZyHdAsMZAQAAoNYwgAJTvCFr5/a/m5KBYcyAbLKY0wRatFF5L55z2+oWqOVLdWMuNdvLNsLVZFtZnClOc0WLViczxetFiyncyviq5pGrAEtsNeQOOZri0cV4g6/dbr0A+v+sDcG4b3mF27SXRNo6U3FzJn2+mQ7NpluboDut6A/E0v1cy8F0LuWDuZhtMd4rKOlp1EuwS70N2BP3c/I99gz3MdvcCMVUrNX2ufFZN9eAOF3NJaCyDh8rwSXLsm7eFnMOcdw7x9F68K3UP20Z4lq8Q5NvrW21l7HqY8uRaO0V2DzOgsVf43d981Pee05Vy9pTvHEBmESjgSInAdvmQazToi/RWRUDAGAB0boIANwrEAA==)
2. Select (Decrypt RSA)
3. Change to Decrypt (ECDH-ES)
4. Notice how some values are added but `kalg` remained unchanged
5. Select Begin-Transaction just to reset the object
6. Select Decrypt (ECDH-ES), notice how the `kalg` value is different

**Comment by schmunk42** [comment]
Try `no_additional_properties: true`.

**Comment by totszwai** [comment]
> Try `no_additional_properties: true`.

No, it doesn't work. It doesn't rebuild the JSON object, so the old value was kept.

**Comment by schmunk42** [comment]
And how about `keep_oneof_values: false`.

@germanbisurgi This version is missing in the demo but available in the playground, right?

**Comment by totszwai** [comment]
> And how about `keep_oneof_values: false`.
> 
> @germanbisurgi This version is missing in the demo but available in the playground, right?

I don't think that's in the demo either:

![image](https://github.com/json-editor/json-editor/assets/4440365/0dd43ce1-f1eb-421a-9557-d10221e419eb)
